docs(hybrid) add status API docs for hybrid data plane (#3024)
Kong/docs.konghq.com: app/gateway-oss/2.5.x/hybrid-mode.md

to the [Plugin Compatibility section][plugin-compat]
of declarative config documentation for more information.

## Readonly Status API endpoints on Data Plane

> Several readonly endpoints from [Admin API](/{{page.kong_version}}/admin-api)

Reviewer: ```suggestion
Several readonly endpoints from the [Admin API](/{{page.kong_version}}/admin-api)
```

> are exposed to [Status API](/{{page.kong_version}}/configuration/#status_listen) on Data Plane.

Reviewer: ```suggestion
are exposed to the [Status API](/{{page.kong_version}}/configuration/#status_listen) on data planes, including the following:
```

> - GET /upstreams/{upstream}/targets/

Reviewer: Should we link to the section of the Admin API where these live?

> - GET /upstreams/{upstream}/health/

Reviewer: ```suggestion
- [GET /upstreams/{upstream}/health/](/{{page.kong_version}}/admin-api/#show-upstream-health-for-node)
```

> - GET /upstreams/{upstream}/targets/all/

Reviewer: ```suggestion
- [GET /upstreams/{upstream}/targets/all/](/{{page.kong_version}}/admin-api/#list-all-targets)
```

- GET /upstreams/{upstream}/targets/{target}

> Please refer to [Upstream objects](/{{page.kong_version}}/admin-api/#upstream-object) for above

Reviewer: ```suggestion
Please refer to [Upstream objects](/{{page.kong_version}}/admin-api/#upstream-object) for more information about the
```
